lib/process_wrapper/process_wrapper.cc

```diff
@@ -0,0 +1,109 @@
+#include <fstream>
+#include <iostream>
+#include <string>
+
+#include "lib/process_wrapper/system.h"
+#include "lib/process_wrapper/utils.h"
+
+using CharType = process_wrapper::System::StrType::value_type;
+
+// Simple process wrapper allowing us to not depend on the shell to run a
+// process to perform basic operations like capturing the output or having
+// the $pwd used in command line arguments or environment variables
+int PW_MAIN(int argc, const CharType* argv[], const CharType* envp[]) {
+  using namespace process_wrapper;
+
+  System::EnvironmentBlock environment_block;
+  // Taking all environment variables from the current process
+  // and sending them down to the child process
+  for (int i = 0; envp[i] != nullptr; ++i) {
+    environment_block.push_back(envp[i]);
+  }
+
+  System::StrType exec_path;
+  System::StrType touch_file;
+  System::StrType stdout_file;
+  System::Arguments arguments;
+  System::Arguments file_arguments;
+  bool subst_pwd = false;
+
+  // Processing current process argument list until -- is encountered
+  // everthing after gets sent down to the child process
+  for (int i = 1; i < argc; ++i) {
+    System::StrType arg = argv[i];
+    if (arg == PW_SYS_STR("--subst-pwd")) {
+      subst_pwd = true;
+    } else {
+      // All argument in the else block have a parameter
+      if (++i == argc) {
+        std::cerr << "process wrapper error: argument \"" << ToUtf8(arg)
+                  << "\" missing parameter." << std::endl;
+        return -1;
+      }
+      if (arg == PW_SYS_STR("--env-file")) {
```

Comment:
Have you tested if the process wrapper works or not when the paths given by `--env-file`, `--arg-file`, etc. are long paths on Windows (>260 characters). 
https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#maximum-path-length-limitation

Reply:
They don't, the wrapper is going to fail, with a message like this one:
process wrapper error: failed to open file: tests/process_wrapper/test_files/lllllllllllllllllllllllllllllllllllllllllllllllooooooooooooooooooooooooooooooooooooooooooooooooonnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg

Reply:
Yeah. If you want the process wrapper to work for long path.
I guess you'll have to convert the paths to a format like `\\?\D:\long\windows\path`.

Reply:
Looking at the state of long path support on windows, if we are operating on paths longer than 260, it is highly probable that other tooling that is used along with the wrapper will break.

Reply:
I see. Can you help add a comment about this tool not supporting long path on Windows? If other people try to use this, they should have the correct expectation.